// @mantine/core/cjs/components/CloseButton/CloseButton.cjs
'use client';
'use strict';

var React = require('react');
var rem = require('../../core/utils/units-converters/rem.cjs');
var getSize = require('../../core/utils/get-size/get-size.cjs');
var createVarsResolver = require('../../core/styles-api/create-vars-resolver/create-vars-resolver.cjs');
require('clsx');
require('@mantine/hooks');
require('../../core/MantineProvider/Mantine.context.cjs');
require('../../core/MantineProvider/default-theme.cjs');
require('../../core/MantineProvider/MantineProvider.cjs');
require('../../core/MantineProvider/MantineThemeProvider/MantineThemeProvider.cjs');
var useProps = require('../../core/MantineProvider/use-props/use-props.cjs');
var useStyles = require('../../core/styles-api/use-styles/use-styles.cjs');
require('../../core/Box/Box.cjs');
var polymorphicFactory = require('../../core/factory/polymorphic-factory.cjs');
require('../../core/DirectionProvider/DirectionProvider.cjs');
var UnstyledButton = require('../UnstyledButton/UnstyledButton.cjs');
var CloseIcon = require('./CloseIcon.cjs');
var CloseButton_module = require('./CloseButton.module.css.cjs');

const defaultProps = {
  variant: "subtle"
};
const varsResolver = createVarsResolver.createVarsResolver((_, { size, radius, iconSize }) => ({
  root: {
    "--cb-size": getSize.getSize(size, "cb-size"),
    "--cb-radius": radius === void 0 ? void 0 : getSize.getRadius(radius),
    "--cb-icon-size": rem.rem(iconSize)
  }
}));
const CloseButton = polymorphicFactory.polymorphicFactory((_props, ref) => {
  const props = useProps.useProps("CloseButton", defaultProps, _props);
  const {
    iconSize,
    children,
    vars,
    radius,
    className,
    classNames,
    style,
    styles,
    unstyled,
    "data-disabled": dataDisabled,
    disabled,
    variant,
    icon,
    ...others
  } = props;
  const getStyles = useStyles.useStyles({
    name: "CloseButton",
    props,
    className,
    style,
    classes: CloseButton_module,
    classNames,
    styles,
    unstyled,
    vars,
    varsResolver
  });
  return /* @__PURE__ */ React.createElement(
    UnstyledButton.UnstyledButton,
    {
      ref,
      ...others,
      unstyled,
      variant,
      disabled,
      mod: { disabled: disabled || dataDisabled },
      ...getStyles("root", { variant, active: true })
    },
    icon || /* @__PURE__ */ React.createElement(CloseIcon.CloseIcon, null),
    children
  );
});
CloseButton.classes = CloseButton_module;
CloseButton.displayName = "@mantine/core/CloseButton";

exports.CloseButton = CloseButton;
