// @mantine/core/cjs/components/Affix/Affix.cjs
'use client';
'use strict';

var React = require('react');
var rem = require('../../core/utils/units-converters/rem.cjs');
var getDefaultZIndex = require('../../core/utils/get-default-z-index/get-default-z-index.cjs');
var createVarsResolver = require('../../core/styles-api/create-vars-resolver/create-vars-resolver.cjs');
require('clsx');
require('@mantine/hooks');
require('../../core/MantineProvider/Mantine.context.cjs');
require('../../core/MantineProvider/default-theme.cjs');
require('../../core/MantineProvider/MantineProvider.cjs');
require('../../core/MantineProvider/MantineThemeProvider/MantineThemeProvider.cjs');
var useProps = require('../../core/MantineProvider/use-props/use-props.cjs');
var useStyles = require('../../core/styles-api/use-styles/use-styles.cjs');
var Box = require('../../core/Box/Box.cjs');
var factory = require('../../core/factory/factory.cjs');
require('../../core/DirectionProvider/DirectionProvider.cjs');
require('../Portal/Portal.cjs');
var OptionalPortal = require('../Portal/OptionalPortal.cjs');
var Affix_module = require('./Affix.module.css.cjs');

const defaultProps = {
  position: { bottom: 0, right: 0 },
  zIndex: getDefaultZIndex.getDefaultZIndex("modal"),
  withinPortal: true
};
const varsResolver = createVarsResolver.createVarsResolver((_, { zIndex, position }) => ({
  root: {
    "--affix-z-index": zIndex?.toString(),
    "--affix-top": rem.rem(position?.top),
    "--affix-left": rem.rem(position?.left),
    "--affix-bottom": rem.rem(position?.bottom),
    "--affix-right": rem.rem(position?.right)
  }
}));
const Affix = factory.factory((_props, ref) => {
  const props = useProps.useProps("Affix", defaultProps, _props);
  const {
    classNames,
    className,
    style,
    styles,
    unstyled,
    vars,
    portalProps,
    zIndex,
    withinPortal,
    position,
    ...others
  } = props;
  const getStyles = useStyles.useStyles({
    name: "Affix",
    classes: Affix_module,
    props,
    className,
    style,
    classNames,
    styles,
    unstyled,
    vars,
    varsResolver
  });
  return /* @__PURE__ */ React.createElement(OptionalPortal.OptionalPortal, { ...portalProps, withinPortal }, /* @__PURE__ */ React.createElement(Box.Box, { ref, ...getStyles("root"), ...others }));
});
Affix.classes = Affix_module;
Affix.displayName = "@mantine/core/Affix";

exports.Affix = Affix;
