// @mantine/core/cjs/components/Indicator/Indicator.cjs
'use client';
'use strict';

var React = require('react');
var rem = require('../../core/utils/units-converters/rem.cjs');
var getSize = require('../../core/utils/get-size/get-size.cjs');
var createVarsResolver = require('../../core/styles-api/create-vars-resolver/create-vars-resolver.cjs');
require('clsx');
var getThemeColor = require('../../core/MantineProvider/color-functions/get-theme-color/get-theme-color.cjs');
var getContrastColor = require('../../core/MantineProvider/color-functions/get-contrast-color/get-contrast-color.cjs');
require('@mantine/hooks');
require('../../core/MantineProvider/Mantine.context.cjs');
require('../../core/MantineProvider/default-theme.cjs');
require('../../core/MantineProvider/MantineProvider.cjs');
require('../../core/MantineProvider/MantineThemeProvider/MantineThemeProvider.cjs');
var useProps = require('../../core/MantineProvider/use-props/use-props.cjs');
var useStyles = require('../../core/styles-api/use-styles/use-styles.cjs');
var Box = require('../../core/Box/Box.cjs');
var factory = require('../../core/factory/factory.cjs');
require('../../core/DirectionProvider/DirectionProvider.cjs');
var getPositionVariables = require('./get-position-variables/get-position-variables.cjs');
var Indicator_module = require('./Indicator.module.css.cjs');

const defaultProps = {
  position: "top-end",
  offset: 0,
  inline: false,
  withBorder: false,
  disabled: false,
  processing: false
};
const varsResolver = createVarsResolver.createVarsResolver(
  (theme, { color, position, offset, size, radius, zIndex, autoContrast }) => ({
    root: {
      "--indicator-color": color ? getThemeColor.getThemeColor(color, theme) : void 0,
      "--indicator-text-color": autoContrast ? getContrastColor.getContrastColor({ color, theme }) : void 0,
      "--indicator-size": rem.rem(size),
      "--indicator-radius": radius === void 0 ? void 0 : getSize.getRadius(radius),
      "--indicator-z-index": zIndex?.toString(),
      ...getPositionVariables.getPositionVariables(position, offset)
    }
  })
);
const Indicator = factory.factory((_props, ref) => {
  const props = useProps.useProps("Indicator", defaultProps, _props);
  const {
    classNames,
    className,
    style,
    styles,
    unstyled,
    vars,
    children,
    position,
    offset,
    inline,
    label,
    radius,
    color,
    withBorder,
    disabled,
    processing,
    zIndex,
    ...others
  } = props;
  const getStyles = useStyles.useStyles({
    name: "Indicator",
    classes: Indicator_module,
    props,
    className,
    style,
    classNames,
    styles,
    unstyled,
    vars,
    varsResolver
  });
  return /* @__PURE__ */ React.createElement(Box.Box, { ref, ...getStyles("root"), mod: { inline }, ...others }, !disabled && /* @__PURE__ */ React.createElement(React.Fragment, null, /* @__PURE__ */ React.createElement(
    Box.Box,
    {
      mod: { "with-label": !!label, "with-border": withBorder, processing },
      ...getStyles("indicator")
    },
    label
  )), children);
});
Indicator.classes = Indicator_module;
Indicator.displayName = "@mantine/core/Indicator";

exports.Indicator = Indicator;
