// @mantine/core/cjs/components/ModalBase/ModalBaseOverlay.cjs
'use client';
'use strict';

var React = require('react');
var Overlay = require('../Overlay/Overlay.cjs');
require('../Transition/transitions.cjs');
var Transition = require('../Transition/Transition.cjs');
var ModalBase_context = require('./ModalBase.context.cjs');
var useModalTransition = require('./use-modal-transition.cjs');

const ModalBaseOverlay = React.forwardRef(
  ({ onClick, transitionProps, style, ...others }, ref) => {
    const ctx = ModalBase_context.useModalBaseContext();
    const transition = useModalTransition.useModalTransition(transitionProps);
    return /* @__PURE__ */ React.createElement(Transition.Transition, { mounted: ctx.opened, ...transition, transition: "fade" }, (transitionStyles) => /* @__PURE__ */ React.createElement(
      Overlay.Overlay,
      {
        ref,
        fixed: true,
        style: [style, transitionStyles],
        zIndex: ctx.zIndex,
        unstyled: ctx.unstyled,
        onClick: (event) => {
          onClick?.(event);
          ctx.closeOnClickOutside && ctx.onClose();
        },
        ...others
      }
    ));
  }
);
ModalBaseOverlay.displayName = "@mantine/core/ModalBaseOverlay";

exports.ModalBaseOverlay = ModalBaseOverlay;
