// @mantine/core/cjs/components/ColorPicker/Swatches/Swatches.cjs
'use client';
'use strict';

var React = require('react');
require('clsx');
require('@mantine/hooks');
require('../../../core/MantineProvider/Mantine.context.cjs');
require('../../../core/MantineProvider/default-theme.cjs');
require('../../../core/MantineProvider/MantineProvider.cjs');
require('../../../core/MantineProvider/MantineThemeProvider/MantineThemeProvider.cjs');
var Box = require('../../../core/Box/Box.cjs');
require('../../../core/DirectionProvider/DirectionProvider.cjs');
var ColorSwatch = require('../../ColorSwatch/ColorSwatch.cjs');
var ColorPicker_context = require('../ColorPicker.context.cjs');

const Swatches = React.forwardRef(
  ({
    className,
    datatype,
    setValue,
    onChangeEnd,
    size,
    focusable,
    data,
    swatchesPerRow,
    ...others
  }, ref) => {
    const ctx = ColorPicker_context.useColorPickerContext();
    const colors = data.map((color, index) => /* @__PURE__ */ React.createElement(
      ColorSwatch.ColorSwatch,
      {
        ...ctx.getStyles("swatch"),
        unstyled: ctx.unstyled,
        component: "button",
        type: "button",
        color,
        key: index,
        radius: "sm",
        onClick: () => {
          setValue(color);
          onChangeEnd?.(color);
        },
        "aria-label": color,
        tabIndex: focusable ? 0 : -1,
        "data-swatch": true
      }
    ));
    return /* @__PURE__ */ React.createElement(Box.Box, { ...ctx.getStyles("swatches"), ref, ...others }, colors);
  }
);
Swatches.displayName = "@mantine/core/Swatches";

exports.Swatches = Swatches;
