// @mantine/core/esm/components/LoadingOverlay/LoadingOverlay.mjs
'use client';
import React from 'react';
import { getDefaultZIndex } from '../../core/utils/get-default-z-index/get-default-z-index.mjs';
import { createVarsResolver } from '../../core/styles-api/create-vars-resolver/create-vars-resolver.mjs';
import 'clsx';
import '@mantine/hooks';
import '../../core/MantineProvider/Mantine.context.mjs';
import '../../core/MantineProvider/default-theme.mjs';
import '../../core/MantineProvider/MantineProvider.mjs';
import { useMantineTheme } from '../../core/MantineProvider/MantineThemeProvider/MantineThemeProvider.mjs';
import { useProps } from '../../core/MantineProvider/use-props/use-props.mjs';
import { useStyles } from '../../core/styles-api/use-styles/use-styles.mjs';
import { Box } from '../../core/Box/Box.mjs';
import { factory } from '../../core/factory/factory.mjs';
import '../../core/DirectionProvider/DirectionProvider.mjs';
import { Loader } from '../Loader/Loader.mjs';
import { Overlay } from '../Overlay/Overlay.mjs';
import '../Transition/transitions.mjs';
import { Transition } from '../Transition/Transition.mjs';
import classes from './LoadingOverlay.module.css.mjs';

const defaultProps = {
  transitionProps: { transition: "fade", duration: 0 },
  overlayProps: { backgroundOpacity: 0.75 },
  zIndex: getDefaultZIndex("overlay")
};
const varsResolver = createVarsResolver((_, { zIndex }) => ({
  root: {
    "--lo-z-index": zIndex?.toString()
  }
}));
const LoadingOverlay = factory((_props, ref) => {
  const props = useProps("LoadingOverlay", defaultProps, _props);
  const {
    classNames,
    className,
    style,
    styles,
    unstyled,
    vars,
    transitionProps,
    loaderProps,
    overlayProps,
    visible,
    zIndex,
    ...others
  } = props;
  const theme = useMantineTheme();
  const getStyles = useStyles({
    name: "LoadingOverlay",
    classes,
    props,
    className,
    style,
    classNames,
    styles,
    unstyled,
    vars,
    varsResolver
  });
  const _overlayProps = { ...defaultProps.overlayProps, ...overlayProps };
  return /* @__PURE__ */ React.createElement(Transition, { transition: "fade", ...transitionProps, mounted: !!visible }, (transitionStyles) => /* @__PURE__ */ React.createElement(Box, { ...getStyles("root", { style: transitionStyles }), ref, ...others }, /* @__PURE__ */ React.createElement(Loader, { ...getStyles("loader"), unstyled, ...loaderProps }), /* @__PURE__ */ React.createElement(
    Overlay,
    {
      ..._overlayProps,
      ...getStyles("overlay"),
      "data-light": true,
      unstyled,
      color: overlayProps?.color || theme.white
    }
  ), /* @__PURE__ */ React.createElement(
    Overlay,
    {
      ..._overlayProps,
      ...getStyles("overlay"),
      "data-dark": true,
      unstyled,
      color: overlayProps?.color || theme.colors.dark[5]
    }
  )));
});
LoadingOverlay.classes = classes;
LoadingOverlay.displayName = "@mantine/core/LoadingOverlay";

export { LoadingOverlay };
